src/katsdpcontroller/generator.py

```diff
@@ -1513,10 +1489,140 @@ def get_baseline_index(a1, a2):
+) -> scheduler.LogicalNode:
+    # Ensure that streams is a sequence, not just an iterable.
+    streams = sorted(streams, key=lambda stream: stream.name)
+
+    n_engines = 1
+    n_substreams = 1
+    base_name = streams[0].name
+    vgpu_group = LogicalGroup(f"vgpu.{base_name}")
+    g.add_node(vgpu_group)
+
+    for stream in streams:
+        stream_sensors: List[Sensor] = [
+            # The timestamps are simply ADC sample counts
+            Sensor(
+                float,
+                f"{stream.name}.scale-factor-timestamp",
+                "Factor by which to divide instrument timestamps to convert to seconds",
+                "Hz",
+                default=stream.adc_sample_rate,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                float,
+                f"{stream.name}.sync-time",
+                "The time at which the digitisers were synchronised. Seconds since the Unix Epoch.",
+                "s",
+                # TODO: Make grandparent property for TARVstream
+                default=sync_time,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                float,
+                f"{stream.name}.bandwidth",
+                "The analogue bandwidth of the digitised band",
+                "Hz",
+                default=acv.bandwidth,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                float,
+                f"{stream.name}.pass-bandwidth",
+                "Bandwidth over which the CBF produces a flat response",
+                "Hz",
+                default=acv.pass_bandwidth,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.n-chans",
+                "The number of frequency channels in this stream's overall output",
+                default=stream.n_chans,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.n-chans-per-substream",
+                "Number of channels in each substream for this data stream",
+                default=stream.n_chans_per_substream,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.n-vengs",
+                "The number of V-engines in the instrument",
+                # TODO: Clarify how this is deciphered. One v-eng to one stream?
+                default=stream.n_vengs,
+                initial_status=Sensor.Status.NOMINAL,
+            ),
+            Sensor(
+                int,
+                f"{stream.name}.veng-out-bits-per-sample",
+                "V-engine output bits per sample. Per number, not complex pair- "
+                "Real and imaginary parts are both this wide",
+                default=stream.bits_per_sample,
```

Comment:
I think you need to add that attribute on GpucbfTiedArrayResampledVoltageStream. It can just be hardcoded to 2 for now.